exporter/elasticsearchexporter/model.go

```diff
@@ -385,7 +352,109 @@ func numberToValue(dp pmetric.NumberDataPoint) (pcommon.Value, error) {
 	return pcommon.Value{}, errInvalidNumberDataPoint
 }
 
-func (m *encodeModel) encodeSpan(resource pcommon.Resource, span ptrace.Span, scope pcommon.InstrumentationScope) ([]byte, error) {
+func (m *encodeModel) encodeResourceOTelMode(document *objmodel.Document, resource pcommon.Resource, resourceSchemaURL string) {
+	resourceMapVal := pcommon.NewValueMap()
+	resourceMap := resourceMapVal.Map()
+	if resourceSchemaURL != "" {
+		resourceMap.PutStr("schema_url", resourceSchemaURL)
+	}
+	resourceMap.PutInt("dropped_attributes_count", int64(resource.DroppedAttributesCount()))
+	resourceAttrMap := resourceMap.PutEmptyMap("attributes")
+	resource.Attributes().CopyTo(resourceAttrMap)
+	stripDataStreamAttributes(resourceAttrMap)
+
+	document.Add("resource", objmodel.ValueFromAttribute(resourceMapVal))
```

Comment:
_nit_: The logic to create a value map and then use `ValueFromAttributes` seems a bit cryptic to me. It eventually adds a sub document but this feels overly complicated. Is it possible to create a `Document` directly (instead of the value map), append the required attributes to the document, add add it as a value attribute?

Reply:
It is currently not possible to do so without modifying objmodel because there is no func to convert a objmodel.Document to objmodel.Value. On top of that Value.kind and Value.doc are unexported fields. Can we defer this to a follow up refactoring PR?

Reply:
SGTM!, Thanks for looking into it.